src/traces/funnelarea/attributes.js

```diff
@@ -0,0 +1,85 @@
+/**
+* Copyright 2012-2019, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var pieAttrs = require('../pie/attributes');
+var sunburstMarker = require('../sunburst/attributes').marker;
+var plotAttrs = require('../../plots/attributes');
+var domainAttrs = require('../../plots/domain').attributes;
+var hovertemplateAttrs = require('../../components/fx/hovertemplate_attributes');
+
+var extendFlat = require('../../lib/extend').extendFlat;
+
+module.exports = {
+    labels: pieAttrs.labels,
+    // equivalent of x0 and dx, if label is missing
+    label0: pieAttrs.label0,
+    dlabel: pieAttrs.dlabel,
+    values: pieAttrs.values,
+    marker: sunburstMarker,
+    text: pieAttrs.text,
+    hovertext: pieAttrs.hovertext,
+
+    scalegroup: extendFlat({}, pieAttrs.scalegroup, {
+        description: [
+            'If there are multiple funnelareas that should be sized according to',
+            'their totals, link them by providing a non-empty group id here',
+            'shared by every trace in the same group.'
+        ].join(' ')
+    }),
+
+    textinfo: extendFlat({}, pieAttrs.textinfo, {
+        flags: ['label', 'text', 'value', 'percent']
+    }),
+
+    hoverinfo: extendFlat({}, plotAttrs.hoverinfo, {
+        flags: ['label', 'text', 'value', 'percent', 'name']
+    }),
+
+    hovertemplate: hovertemplateAttrs({}, {
+        keys: ['label', 'color', 'value', 'percent', 'text']
+    }),
+
+    textfont: pieAttrs.textfont,
+    insidetextfont: pieAttrs.insidetextfont,
+    outsidetextfont: pieAttrs.outsidetextfont,
```

Comment:
There's no way to set `textposition: 'outside'` at the moment for `funnelarea`, so we should :hocho: this.

Reply:
done in https://github.com/plotly/plotly.js/pull/3876/commits/32c96d2926d41c7b557635fc1da8dcfb765f5eaf